edgelet/edgelet-kube/src/module/create.rs

```diff
@@ -51,9 +71,51 @@ where
         })
 }
 
+fn get_module_owner<T, S>(
+    runtime: &KubeModuleRuntime<T, S>,
+    module_name: &str,
+) -> impl Future<Item = KubeModuleOwner, Error = Error>
+where
+    T: TokenSource + Send + 'static,
+    S: Service + Send + 'static,
+    S::ReqBody: From<Vec<u8>>,
+    S::ResBody: Stream,
+    Body: From<S::ResBody>,
+    S::Error: Fail,
+    S::Future: Send,
+{
+    let module_name = module_name.to_string();
+    runtime
+        .client()
+        .lock()
+        .expect("Unexpected lock error")
```

Comment:
Be more specific on this error message. What lock is failing?

Reply:
It is the global client lock. Staying within error message convention for now.

Reply:
normally it should not happen in a way how we use kube-client, but rust requires us to handle this explicitly

Reply:
@massand  - Okay, but why can't the message be "Global client lock failure" :) Would help with debugging if it ever were to happen, no?

Reply:
The stack trace is plenty for identifying one of 8 places this lock error could occur in this module.